test/forms/inputs/hidden-input.test.js

```diff
@@ -1,29 +1,25 @@
 import React from 'react'
-import { mount } from 'enzyme'
 import { HiddenInput } from '../../../src/'
+import { render, screen } from '@testing-library/react'
 
 const name = 'my.hidden.input'
 const value = 'foo'
 const onChange = jest.fn()
 const props = { input: { name, value, onChange }, meta: {} }
 
 test('HiddenInput renders an input', () => {
-  const wrapper = mount(<HiddenInput {...props} />)
-  expect(wrapper.find('input').exists()).toEqual(true)
+  render(<HiddenInput {...props} />)
+  const input = screen.getByRole('textbox', { hidden: true })
+
+  expect(input).toBeInTheDocument()
 })
 
 test('HiddenInput renders a div the correct styles', () => {
-  const wrapper = mount(<HiddenInput {...props} />)
-  expect(wrapper.find('div').first().props().style).toHaveProperty(
-    'position',
-    'absolute'
-  )
-  expect(wrapper.find('div').first().props().style).toHaveProperty(
-    'left',
-    -9999
-  )
-  expect(wrapper.find('div').first().props().style).toHaveProperty(
-    'visibility',
-    'hidden'
-  )
+  render(<HiddenInput {...props} data-testid="test" />)
+
+  const wrapper = screen.getByTestId('hidden-input-wrapper')
+
+  expect(wrapper).toHaveStyle('visibility: hidden')
+  expect(wrapper).toHaveStyle('left: -9999px')
+  expect(wrapper).toHaveStyle('position: absolute')
```

Comment:
I'm not a huge fan of this test. Can we replace it with a combination of toBeInTheDocument and [toBeVisible](https://github.com/testing-library/jest-dom#tobevisible)? And in that case would we _need_ to test the div with the testid or could we keep this assertion on the element with the `textbox` role?

Reply:
Yes, 100%. I removed this test and added the assertion that the input is in the DOM but not visible. It shouldn't matter how we accomplish that in CSS.